source/dagger/gameplay/team_game/team_game_collisions.cpp

```diff
@@ -0,0 +1,156 @@
+                if (state == MovementState::MOVEABLE) pos_.x -= p.x + size.x - p2.x;// +0.1f;
+
+                collisionInfo.collisionSideOther = CollisionSide::LEFT;
+                if (other_.state == MovementState::MOVEABLE) posOther_.x -= p2.x - p.x - size.x;// - 0.1f;
+            }
+        }
+
+
+        if (p.x <= (p2.x + other_.size.x) && p.x >= p2.x)
+        {
+            if ((abs(p.x - p2.x - other_.size.x) < abs(p.y - p2.y - other_.size.y)) && ((abs(p.x - p2.x - other_.size.x) < abs(p.y + size.y - p2.y))))
+            {
+                // bumped into other_ from the right side
+                collisionInfo.collisionSide = CollisionSide::LEFT;
+                if (state == MovementState::MOVEABLE) pos_.x -= p.x - p2.x - other_.size.x;// -0.1f;
+
+                collisionInfo.collisionSideOther = CollisionSide::RIGHT;
+                if (other_.state == MovementState::MOVEABLE) posOther_.x -= p2.x + other_.size.x - p.x;// +0.1f;
+            }
+        }
+
+
+        if ((p.y + size.y) >= p2.y && (p2.y + other_.size.y) >= (p.y + size.y))
+        {
+            if ((abs(p.y + size.y - p2.y) < abs(p.x - p2.x - other_.size.x)) && ((abs(p.y + size.y - p2.y) < abs(p.x + size.x - p2.x))))
+            {
+                // bumped into other_ from the bottom side
+                collisionInfo.collisionSide = CollisionSide::TOP;
+                if (state == MovementState::MOVEABLE) pos_.y -= p.y + size.y - p2.y + 0.1f;
+
+                collisionInfo.collisionSideOther = CollisionSide::BOTTOM;
+                if (other_.state == MovementState::MOVEABLE) posOther_.y -= p2.y - p.y - size.y - 0.1f;
+            }
+        }
+
+        if (p.y <= (p2.y + other_.size.y) && p.y >= p2.y)
+        {
+            if ((abs(p.y - p2.y - other_.size.y) < abs(p.x - p2.x - other_.size.x)) && ((abs(p.y - p2.y - other_.size.y) < abs(p.x + size.x - p2.x))))
+            {
+                // bumped into other_ from the top side
+                collisionInfo.collisionSide = CollisionSide::BOTTOM;
+                if (state == MovementState::MOVEABLE) pos_.y -= p.y - p2.y - other_.size.y - 0.1f;
+
+                collisionInfo.collisionSideOther = CollisionSide::TOP;
+                if (other_.state == MovementState::MOVEABLE) posOther_.y -= p2.y + other_.size.y - p.y + 0.1f;
+            }
+        }
+
+    }
+    else
+    {
+        collisionInfo.hasCollided = false;
+        collisionInfo.collisionSide = CollisionSide::NONE;
+        collisionInfo.collisionSideOther = CollisionSide::NONE;
+    }
+
+    return collisionInfo;
+}
+
+void PlatformerCollisionSystem::LimitPlayerMovement(Collider& collision_)
+{
+    if (!collision_.listOfEntities.empty())
+    {
+        for (int i = 0; i < collision_.listOfEntities.size(); i++)
+        {
+            collision_.canGoRight = (collision_.listOfCollisionSides[i] == CollisionSide::RIGHT) ? false : collision_.canGoRight;//character_.canGoRight;
```

Comment:
We should word wrap these lines to maintain readability

Reply:
And just a question, since this is a non-issue for me, how would you go about wrapping said line?

Reply:
I wrapped what i could, but i don't want to wrap further since i think that'll damage the readability in my opinion